esmvaltool/_recipe.py

```diff
@@ -751,10 +745,10 @@ def _get_preprocessor_task(variables, profiles, config_user, task_name):
                 config_user,
                 name=derive_name)
             derive_tasks.append(task)
-
-    # Add CMOR info
-    for variable in variables:
-        _add_cmor_info(variable)
+    else:
+        # Add CMOR info
```

Comment:
Maybe this can be done before the `if variable.get('derive')` block and then the line ` _add_cmor_info(variable, derive=True)` can be removed from the function `_get_derive_input_variables`, because now this function has the unexpected side effect of changing `variables`.

Reply:
Done